File: WebRouterApp/ClientApp/src/app/trading/models/messages/orders-changed.message.ts
import { IOrder } from '../order';
import { Guid } from '../../../shared/types/guid';
import { TradingMessageTags } from './trading-message-tags';

export interface IOrdersChangedMessage {
  tag: TradingMessageTags.OrdersChanged;
  traderId: Guid;
  changeSet: IOrdersChangeSet;
}

export interface IOrdersChangeSet {
  added: IOrder[];
  removed: IOrder[];
}

File: WebRouterApp/Features/Trading/Application/MessageParts/OrdersChangedMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace WebRouterApp.Features.Trading.Application.MessageParts
{
    public class OrdersChangedMessage : ITradingMessage
    {
        public OrdersChangedMessage(Guid traderId, OrdersChangeSet changeSet)
        {
            TraderId = traderId;
            ChangeSet = changeSet;
        }

        public string Tag => "Trading.OrdersChanged";

        public Guid TraderId { get; }
        public OrdersChangeSet ChangeSet { get; }
    }

    public class OrdersChangeSet
    {
        public OrdersChangeSet(
            IEnumerable<OrderModel> added, 
            IEnumerable<OrderModel> removed)
        {
            Added = added.ToList();
            Removed = removed.ToList();
        }

        public IReadOnlyList<OrderModel> Added { get; }
        public IReadOnlyList<OrderModel> Removed { get; }
    }

    public static class OrdersChangeSetExtensions
    {
        public static bool Any(this OrdersChangeSet changeSet)
            => changeSet.Added.Any() ||
               changeSet.Removed.Any();

        public static bool IsEmpty(this OrdersChangeSet changeSet)
            => !changeSet.Any();
    }
}